tiledb/sm/query/global_order_writer.cc

```diff
@@ -1001,58 +1001,25 @@ Status GlobalOrderWriter::prepare_full_tiles_var(
             name, &((*tiles)[i]), &((*tiles)[i + 1]), &((*tiles)[i + 2])));
 
     // Handle last tile (it must be either full or empty)
+    uint64_t tile_idx = 0;
     if (last_tile_cell_idx == cell_num_per_tile) {
       last_var_offset = 0;
       (*tiles)[0].swap(last_tile);
       (*tiles)[1].swap(last_tile_var);
       if (nullable) {
         (*tiles)[2].swap(last_tile_validity);
       }
+      tile_idx += t;
     } else {
       assert(last_tile_cell_idx == 0);
     }
 
     // Write all remaining cells one by one
     uint64_t current_tile_cell_idx = 0;
-    uint64_t tile_idx = 0;
-    if (coord_dups.empty()) {
-      for (uint64_t i = 0; i < cell_num_to_write;
-           ++cell_idx, ++i, ++current_tile_cell_idx) {
-        if (current_tile_cell_idx == cell_num_per_tile) {
-          (*tiles)[tile_idx + 1].final_size(last_var_offset);
-          current_tile_cell_idx = 0;
-          last_var_offset = 0;
-          tile_idx += t;
-        }
-
-        // Write offset.
-        RETURN_NOT_OK((*tiles)[tile_idx].write(
-            &last_var_offset,
-            current_tile_cell_idx * sizeof(last_var_offset),
-            sizeof(last_var_offset)));
-
-        // Write var-sized value(s).
-        auto buff_offset =
-            prepare_buffer_offset(buffer, cell_idx, attr_datatype_size);
-        uint64_t var_size = (cell_idx == cell_num - 1) ?
-                                *buffer_var_size - buff_offset :
-                                prepare_buffer_offset(
-                                    buffer, cell_idx + 1, attr_datatype_size) -
-                                    buff_offset;
-        RETURN_NOT_OK((*tiles)[tile_idx + 1].write_var(
-            buffer_var + buff_offset, last_var_offset, var_size));
-        last_var_offset += var_size;
-
-        // Write validity value(s).
-        if (nullable)
-          RETURN_NOT_OK((*tiles)[tile_idx + 2].write(
-              buffer_validity + cell_idx,
-              current_tile_cell_idx * constants::cell_validity_size,
-              constants::cell_validity_size));
-      }
-    } else {
-      for (uint64_t i = 0; i < cell_num_to_write; ++cell_idx, ++i) {
-        if (coord_dups.find(cell_idx) == coord_dups.end()) {
+    if (cell_num_to_write != 0) {
```

Comment:
This diff after this is statement is incorrect, the code below has not changed until my next comment.